### PR TITLE
Do not cache a file miss or corrupt file.

### PR DESCRIPTION
**Pull request by pmclanahan**
This could have been caused by a temporary condition,
and you'd want to recover as quickly as possible.

### CODE REVIEW

**Comment by pmclanahan** [comment]
@jgmize r? I think this is the first step.

File: tests.py [hunk 1]
```diff
@@ -151,18 +151,28 @@ def test_cache_delete():
...
     ok_(isinstance(pd.the_dude, defaultdict))
     ok_(isinstance(pd.the_dude, defaultdict))
-    pd.delete_cache('the_dude')
+    eq_(open_mock.call_count, 2)
+
+
+@patch('codecs.open')
+@patch('json.load')
+def test_no_cache_corrupt_files(load_mock, open_mock):
+    """The fact that a file doesn't exist should not be cached."""
```
Comment (jgmize) [inline tests.py]: NIT: inaccurate docstring

**Comment by jgmize** [comment]
r+wc

**Comment by pmclanahan** [comment]
Good catch. Fixed. Thanks Josh.

**Pull request merged**